### PR TITLE
Report Vulkan errors to the report view

### PR DESCRIPTION
**Pull request by Qining**
Use `vkGetImageMemoryRequirements` as an example.

### CODE REVIEW

File: gapis/api/vulkan/externs.go [hunk 1]
```diff
@@ -1115,3 +1117,24 @@ func bindSparse(ctx context.Context, s *api.GlobalState, binds *QueuedSparseBind
...
+	var issue replay.Issue
+	issue.Command = e.cmdID
+	issue.Severity = service.Severity_ErrorLevel
+	switch err := err.(type) {
+	case *ERR_INVALID_HANDLE:
+		issue.Error = fmt.Errorf("Invalid %s: %v", err.HandleType, err.Handle)
+	case *ERR_NULL_POINTER:
+		issue.Error = fmt.Errorf("Null pointer of %s", err.PointerType)
+	case *ERR_UNSUPPORTED_EXTENSION:
+		issue.Error = fmt.Errorf("Unsupported extension: %s", err.Name)
```
Comment (AWoloszyn) [inline gapis/api/vulkan/externs.go]: I would probably make this a warning.
To me ERR_UNSUPPORTED_EXTENSION would be if you use an extension that the driver does not provide.

In THIS case we are talking about an extension that GAPID does not understand. Which is not an issue with the application itself.

File: gapis/api/vulkan/errors.api [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  onVkError(new!ERR_NULL_POINTER(pointerType: pointerType))
+  abort
+}
+
+sub void vkErrorUnsupportedExtension(const char* name) {
+  onVkError(new!ERR_UNSUPPORTED_EXTENSION(name: as!string(name)))
+  // Continue the mutation as this may not cause problem.
+}
+
+class ERR_INVALID_HANDLE {
```
Comment (ben-clayton) [inline gapis/api/vulkan/errors.api]: These should be generated by the stringtable system. Declare them in `en-us.stb.md` and `import "../../messages/messages.api"`

Comment (Qining) [reply gapis/api/vulkan/errors.api]: Importing the `../../messages/message.api` causes an error on the CPP side:
```
.../gapid/gapii/cc/vulkan_types.h:1983:27: error: redefinition of 'SEVERITY_DEBUG'
```

`Severity` is defined in both `gles_types.h` and `vulkan_types.h`.

Comment (Qining) [reply gapis/api/vulkan/errors.api]: Is there a way to add a namespace for the types defined in `gles_types.h` and `vulkan_types.h`. That will solve the problem.

**Comment by Qining** [comment]
@ben-clayton Reverted back to defining Vulkan's error message in vulkan/errors.api, and using extern function to receive the messages.
@AWoloszyn Changed from `UnsupportedExtention` to `UnrecognizedExtension`

If this gets in, #1371 should be discarded.

**Pull request merged**